tests/algorithms/test_pgd.py

```diff
@@ -0,0 +1,159 @@
+"""Tests for the proximal gradient descent."""
+
+import torch
+from mrpro.algorithms.optimizers import pgd
+from mrpro.data.SpatialDimension import SpatialDimension
+from mrpro.operators import FastFourierOp, LinearOperatorMatrix, ProximableFunctionalSeparableSum
+from mrpro.operators.functionals import L1Norm, L2NormSquared
+from mrpro.phantoms import EllipsePhantom
+
+
+def test_pgd_convergence_fft_example() -> None:
+    """Test if the proximal gradient descent algorithm converges for the
+    problem min_x 1/2 ||Fx - target||^2 + ||x||_1,
+    with F being the Fourier Transform."""
+    dim = SpatialDimension(x=100, y=100, z=1)
+    ellipse = EllipsePhantom()
+    image = ellipse.image_space(dim)
+    fft = FastFourierOp()
+    (kspace,) = fft(image)
+
+    l2 = L2NormSquared(target=kspace, divide_by_n=True)
+    f = l2 @ fft
+    g = L1Norm(divide_by_n=True)
+
+    initial_value = torch.ones_like(image)
+    pgd_solution = pgd(
+        f=f,
+        g=g,
+        initial_value=initial_value,
+        stepsize=0.5,
+        reg_parameter=0.01,
+        max_iterations=200,
+        backtrack_factor=1.0,
+    )
+
+    assert f(*pgd_solution)[0] + g(*pgd_solution)[0] < f(*initial_value)[0] + g(*initial_value)[0]
```

Comment:
I think this does not imply that the algorithm has converged... this would (most probably) be even the case after a few iterations (if fista monotonically decreases the objective functional - which I am not 100% sure about).
I would suggest to replace the test to check if fista has calculated the true solution to the problem. For certain this problem, the solution would be given by simple soft-thresholding, see also here:

https://github.com/PTB-MR/mrpro/blob/05c259451a62789188492e271f92a8138d780534/tests/algorithms/test_pdhg.py#L101